### PR TITLE
Default: Add bush stem and leaves nodes 

### PR DESCRIPTION
**Pull request by paramat**
To allow mapgen bushes in green-grass and savanna grasslands.
Nodes for a generic bush and an acacia bush.
Stem nodes are craftable to a single wood node to provide a small
amount of wood resource in grasslands.
Fuel times are that of corresponding 'wood' nodes, 1/4 that of
corresponding tree nodes.
No leafdecay to enable use as hedges or without a nearby tree trunk.
Uses 'default leaves simple' texture for extra visual thickness.
//////////////////////////////////////////////////

screenshot_20161026_005801

screenshot_20161026_023039

^ Side stems actually touch cube faces precisely, the visual gap is due to the 44 degree rotation of plantlike. The central stem has been chopped to only penetrate the top leaf node by < 1 pixel.

screenshot_20161026_010137

^ Leaf nodes have visual thickness to enable use on their own as hedges, maze hedges etc.

screenshot_20161026_010539

screenshot_20161026_023236

^ Bush stem adds darkness and stem colour to the bush centre

### CODE REVIEW

**Comment by tobyplowy** [comment]
@paramat The leaf Texture Looks ok in my opinion allthow the jungle should have a More Dance version

**Comment by Fixer-007** [comment]
It looks funny, it is hard to form an opinion about it.

**Comment by paramat** [comment]
Hm it looks out of place in savanna, perhaps we should have a seperate 'acacia_bush_leaves' node for that?

**Comment by tenplus1** [comment]
This is what I envision bushels to be like if added:#
Bushels

**Comment by tobyplowy** [comment]
I agree With @tenplus1 I think The wood inside The budhes is not so good looking

**Comment by tobyplowy** [comment]
@paramat I think Every biome shuld have its own bush

**Comment by Fixer-007** [comment]
Yes, that trunk is bothering me too, maybe use stick instead? or immerse trunk in soil...

**Comment by paramat** [comment]
The trunk is meant for providing some tool-building wood in grasslands. The new biomes mean even more chance of being far from forests.

**Comment by paramat** [comment]
Updated, see first post.
Now 2 nodes, both use existing textures.

**Comment by tobyplowy** [comment]
@paramat The bushes are new nodes arend they Maybe if you break Them they turn into sticks so you Can craft them into planks?

**Comment by paramat** [comment]
Updated after discussion http://irc.minetest.net/minetest-dev/2016-10-14#i_4724491
'bush stem' nodes are added that are craftable to 1 wood node to provide a limited wood resource in grasslands (4 bushes needed to create a wood pick).
Stem nodes are identical to 'tree' nodes and use existing textures.

**Comment by Ferk** [comment]
I don't understand. Why use a new stem node instead of a tree node if they are identical?
From the IRC dialog I get the impression @sfan5 meant to use a plantlike node instead of a blocky trunk.

**Comment by paramat** [comment]
Yeah to provide 1 wood instead of 4.
It could be a plantlike drawtype.
I also just thought that this stem could have bark texture on all 6 sides, to darken it's appearence, this would also distinguish it from the tree versions.

**Comment by tobyplowy** [comment]
@paramat Maybe the Wood Block could be Underneath The bush in the ground (on the Same level as grass blocks)

**Comment by sfan5** [comment]
@Ferk I indeed suggested to use a `plantlike` drawtype as bushes are not as dense as normal trees.

**Comment by Ferk** [comment]
Another alternative for the stem would be to make it a nodebox for a much thinner trunk (25% the size as it gives 25% the wood), this way the tree texture can still be reused in all sides (just like fences use wood), and it might make for an interestingly different node.

**Comment by paramat** [comment]
I considered a nodebox but would prefer plantlike for render speed, plantlike is more suitable too for this use, the stem is mostly hidden so a fancy appearence is mostly wasted. I'll work on a plantlike stem.

**Comment by sofar** [comment]
Could use the `meshoptions` to give it a different shape as well?

**Comment by paramat** [comment]
Indded.

**Comment by DS-Minetest** [comment]
wouldn't it be better to make the stem as no real stem but as leaf like node, that got some twinks on its sides and on bottom?
or make a mesh with leaf texture on side and some twinks in this 1m³ block looking out and stabbing into the leaf nodes around
and then generate some leaves around it

**Comment by paramat** [comment]
screenshot_20161025_072647

bush_3_stems128

Working on plantlike stems, much better, these will directly drop 1 'wood' node.

**Comment by tobyplowy** [comment]
@paramat Looks good :) although it would probably Look Better if it Inside a leaf block

**Comment by DS-Minetest** [comment]
ok, i was false, parmat's plantlike stem is really nice

**Comment by sofar** [comment]
I've been avoiding commenting on this thread for a while, because I can't seem to get warmed up to where this is going...

TBH I wish we would just redo leaves entirely and avoid having to add stems:
- leaves dug by hand drop 0-1 sticks
- leaves dug with scissors drop leaves

Then we could just place leaves as bushes in the landscape and players would only ever get sticks out of them, while making the ornamental placement of leaves a bit harder (requiring a scissor tool made out of e.g. copper)

I like doing more without new nodes, and that would be a simple way of doing it. The default mtg can really use a default shear/scissor tool as well (could even just make it a knife, whatever) since so many mods rely on them already.

**Comment by paramat** [comment]
We need some wood dropped in grasslands. and bushes can't have leafdecay.

**Comment by sofar** [comment]
just place them with a different param2?

**Comment by sofar** [comment]
Erg my bad, coordination issues :)

**Comment by paramat** [comment]
Bush leaves need a thicker leaves texture too, these use 'leaves_simple'.

**Comment by paramat** [comment]
@sfan5 you were correct, the first attempt was too thick, especially when enalarged to fill the cube.
Updated, see first post.

**Comment by sfan5** [comment]
Stem texture looks good, especially in combination with leaves :+1:

File: mods/default/nodes.lua [hunk 1]
```diff
@@ -1216,6 +1221,55 @@ for i = 2, 5 do
...
+minetest.register_node("default:bush_leaves", {
+	description = "Bush Leaves",
+	drawtype = "allfaces_optional",
+	waving = 1,
+	visual_scale = 1.3,
+	tiles = {"default_leaves_simple.png"},
+	paramtype = "light",
+	groups = {snappy = 3, flammable = 2, leaves = 1},
+	sounds = default.node_sound_leaves_defaults(),
+})
```
Comment (sfan5) [inline mods/default/nodes.lua]: Should these be usable as fuel?

**Comment by DS-Minetest** [comment]
what about using texture modifier to make the bush leaves thicker
i thought of something like this:
`default_leaves.png^(default_leaves.png^[transformFX)`

**Comment by sofar** [comment]
Have you considered using `meshoptions` and set `param2 = 1` ? That may look good as well, and may work better when there's no leaves diagonally.

I'm somewhat defrosting to this concept, the latest screenshots posted at the top don't look so bad at all.

**Comment by tobyplowy** [comment]
@paramat it Looks Good how it is right now

File: mods/default/nodes.lua (hunk 1 above)
Comment (paramat) [inline mods/default/nodes.lua]: Fuel time is defined for group:leaves.

**Comment by paramat** [comment]
Yes i've considered the + shape, as you wrote because diagonal leaves are not always present. I might prefer diagonal because it's more consistent with the widespread use of plantlike, i like the way it adds some 'diagonality' to the bush, i like how it allows a scaled-up stem, which helps add more darkness to the centre and makes the stems thicker.

There's time to reconsider, for now though i need these in to work on biomes for release so i'll merge.

**Comment by paramat** [comment]
048aaa2616e25ee15cd04a0d50f1a61ff2bdcd38

**Comment by tobyplowy** [comment]
@paramat How about Jungle bushes (I think the jungle floor looks a bit too naked, this would help) it's just a question and sort of a request :P

**Comment by paramat** [comment]
Nope, i opposed this in sofar's PR.
Rainforest is already very dense, we could increase the density of junglegrass.
In reality the rainforest floor is not particularly overgrown.

**Comment by tobyplowy** [comment]
@paramat increaseing the density of junglegrass is also good i guss

**Pull request closed without merging**